bookkeeper-server/src/test/java/org/apache/bookkeeper/client/MockLedgerHandle.java

```diff
@@ -108,7 +108,7 @@ public void run() {
                 final Queue<LedgerEntry> seq = new ArrayDeque<LedgerEntry>();
                 long entryId = firstEntry;
                 while (entryId <= lastEntry && entryId < entries.size()) {
-                    seq.add(new LedgerEntry(entries.get((int) entryId++)));
+                    seq.add(new LedgerEntry(entries.get((int) entryId++).duplicate()));
```

Comment:
`LedgerEntryImpl` is a new api implementation, we are converting it to the legacy "LedgerEntry" implementation. so better to written as following:

```
try (LedgerEntryImpl duplicatedEntry = entries.get((int) entryId++).duplicate()) {
   seq.add(new LedgerEntry(duplicatedEntry));
}

```

Reply:
Already merged. If I'm in the area again, ill change